server/src-lib/Hasura/GraphQL/Transport/WebSocket/Server.hs

```diff
@@ -68,23 +70,49 @@ $(J.deriveToJSON
                    }
   ''WSEvent)
 
+-- extra websocket event info
+data WSEventInfo
+  = WSEventInfo
+  { _wseiQueryExecutionTime :: !(Maybe Double)
+  , _wseiResponseSize       :: !(Maybe Int64)
+  } deriving (Show, Eq)
+$(J.deriveToJSON
+  J.defaultOptions { J.fieldLabelModifier = J.snakeCase . drop 5
+                   , J.omitNothingFields = True
+                   }
+  ''WSEventInfo)
+
+
 data WSLog
   = WSLog
   { _wslWebsocketId :: !WSId
   , _wslEvent       :: !WSEvent
+  , _wslMetadata    :: !(Maybe WSEventInfo)
   } deriving (Show, Eq)
-$(J.deriveToJSON (J.aesonDrop 4 J.snakeCase) ''WSLog)
+$(J.deriveToJSON
+  J.defaultOptions { J.fieldLabelModifier = J.snakeCase . drop 4
+                   , J.omitNothingFields = True
+                   }
+  ''WSLog)
 
 instance L.ToEngineLog WSLog L.Hasura where
   toEngineLog wsLog =
     (L.LevelDebug, L.ELTInternal L.ILTWsServer, J.toJSON wsLog)
 
+data WSQueueResponse
+  = WSQueueResponse
+  { _wsqrMessage   :: !BL.ByteString
+  , _wsqrEventInfo :: !(Maybe WSEventInfo)
+  -- ^ extra metadata that we use for other actions, such as print log
+  -- we don't want to inlcude them into websocket message payload
```

Comment:
Please use [Haddock comment syntax](https://haskell-haddock.readthedocs.io/en/latest/markup.html#documenting-parts-of-a-declaration) for comments like there where it makes sense. (I’ve made the change here myself, since it’s a small thing, but just so you know in the future.)